src/main/java/serenitylabs/tutorials/vetclinic/playingball/model/SportsSchedule.java

```diff
@@ -0,0 +1,21 @@
+package serenitylabs.tutorials.vetclinic.playingball.model;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+
+/**
+ * Created by kimahale on 9/8/2016.
+ */
+public class SportsSchedule {
+
+    GameSchedule sportsSchedule = new SchoolSportsSchedule();
+    GameSchedule recreationalSportScheduled = new RecreationalSportsSchedule();
+
+    public PlayBall forDate(LocalDate currentDay) {
+        if (currentDay.getDayOfWeek()== DayOfWeek.WEDNESDAY)
+            return sportsSchedule.forGameOnByWeek(currentDay);
```

Comment:
If you call forGameOn() in both cases, you can just use the current day to pick which calendar to use, and not need to call different methods. Then you can use a map of the days of the week to GameSchedule and have a solution that doesn't need an if-else condition at all.